### ISSUE TITLE
`Rails/DynamicFindBy` false-positives on `Gem::Specification.find_by_name`

### ISSUE THREAD

**Issue by tobypinder** [rubocop-hq/rubocop#4967]
`Gem::Specification.find_by_name` clearly looks like `ActiveRecord` to this cop but isn't.

False positives are inevitable and [whitelist functionality has been explained before](https://github.com/bbatsov/rubocop/issues/3758), but given this class is in rubygems itself it would feel prudent to add [`Gem::Specification#find_by_name`](https://ruby-doc.org/stdlib-2.3.4/libdoc/rubygems/rdoc/Gem/Specification.html#method-c-find_by_name) and [`Gem::Specification#find_by_path`](https://ruby-doc.org/stdlib-2.3.4/libdoc/rubygems/rdoc/Gem/Specification.html#method-c-find_by_path) to a _default_ or even hardcoded whitelist inside the cop, as I cannot think of any scenario other than a namespace collision where detecting these as activerecord is expected behaviour.

--------

## Expected behavior

No offences found.

## Actual behavior

`Rails/DynamicFindBy: Use 'find_by' instead of dynamic 'find_by_name'. (Rails/DynamicFindBy)`

## Steps to reproduce the problem

```ruby
Gem::Specification.find_by_name('loremipsum')
```

## RuboCop version

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.5 x86_64-linux)
```

**Comment by tobypinder** [comment]
I see the comments from the @koic's attempt at resolving this. Would it be possible to expand the `Whitelist` functionality from `method` to `ClassName#instance_method` / `ClassName.class_method`? 

Doesn't deal with subclasses as @koic attempted but surely the class name itself is available to the AST, and we're not particularly worried about `Gem::Specification` getting subclassed here? Failing that can we whitelist whole classes under a different key?

**Comment by bbatsov** [comment]
We're in the process of moving all Rails-related functionality to a standalone library (gem) named `rubocop-rails`. Please, migrate this issue to its issue tracker https://github.com/rubocop-hq/rubocop-rails/issues/
